Minio.Functional.Tests/FunctionalTest.cs

```diff
@@ -2329,21 +2327,27 @@ internal async static Task GetObject_Test2(MinioClient minio)
             try
             {
                 await Setup_Test(minio, bucketName);
-                try
-                {
-                    GetObjectArgs getObjectArgs = new GetObjectArgs()
-                                                            .WithBucket(bucketName)
-                                                            .WithObject(objectName)
-                                                            .WithFile(fileName);
-                    await minio.GetObjectAsync(getObjectArgs);
+            }
+            catch (Exception ex)
+            {
+                await TearDown(minio, bucketName);
+                new MintLogger("GetObject_Test2", getObjectSignature, "Tests for non-existent GetObject", TestStatus.FAIL, (DateTime.Now - startTime), ex.Message, ex.ToString(), args:args).Log();
+                throw ex;
+            }
 
-                }
-                catch (ObjectNotFoundException ex)
-                {
-                    Assert.AreEqual(ex.ServerMessage, "Not found.");
-                }
+            try
+            {
+                GetObjectArgs getObjectArgs = new GetObjectArgs()
+                                                        .WithBucket(bucketName)
+                                                        .WithObject(objectName)
+                                                        .WithFile(fileName);
+                await minio.GetObjectAsync(getObjectArgs);
                 new MintLogger("GetObject_Test2", getObjectSignature, "Tests for non-existent GetObject", TestStatus.PASS, (DateTime.Now - startTime), args:args).Log();
             }
+            catch (ObjectNotFoundException ex)
```

Comment:
what does this exception do?

Reply:
Checks `GetObjectAsync` for an object that doesn't exist, verifies that the exception is indeed thrown.

Reply:
if that is the case, then `Asset.AreEqual` is not needed, you can simply log `PASS` in the log. The logger statement after `GetObjectAsync` should not be present. You should simply throw an exception which will be caught in the exception handler.

Reply:
Changed.